src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String INDEX_ERROR = "Index is wrong";
+    private int listSize = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private void connection(Node<T> node) {
+        if (head == null) {
+            head = tail = node;
+        } else {
+            tail.next = node;
+            tail = node;
+        }
+    }
+
+    private Node<T> stepBystep(int index) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < index; i++) {
+            currentNode = currentNode.next;
+        }
+        return currentNode;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        connection(newNode);
+        listSize++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > listSize || index < 0) {
+            throw new IndexOutOfBoundsException(INDEX_ERROR);
+        }
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == 0) {
+            newNode.next = head;
+            head = newNode;
+            listSize++;
+            return;
+        }
+        Node<T> currentNode = head;
+        for (int i = 0; i < index - 1; i++) {
+            currentNode = currentNode.next;
+        }
+        newNode.next = currentNode.next;
+        currentNode.next = newNode;
+        listSize++;
+        if (newNode.next == null) {
+            tail = newNode;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
+            throw new NullPointerException("List is null");
+        }
+        for (int i = 0; i < list.size(); i++) {
+            T element = list.get(i);
```

Comment:
Consider using _for-each_ loop, as `get(i)` operation sometime could be really expensive, especially in `LinkedList` case.